### ISSUE TITLE
Colon-space error "Could not save page"

### ISSUE THREAD

**Issue by bitinerant** [zim-desktop-wiki/zim-desktop-wiki#696]
Leaving certain pages brings up a dialog box "Could not save page". The Cancel button shows traceback 1, below. The Discard Changes button does nothing. The Save Copy button, followed by Save, shows trackeback 2, below.

To reproduce this issue:
* create a link to a new page and click on the link to open the new page
* press Backspace enough times to completely delete the title, if any, so that what you type is in the normal font and not a title
* type: `a: b` (that's a-colon-space-b)
* press the back arrow to return to the previous page

Every time I've done the above steps, I get the errors described. If there are other lines after the `a: b`, no error dialog appears, but the first line of text is silently deleted.

I'm running Ubuntu 18.04, running Zim from source (`git clone ...` and `git checkout 98f8284` and `./zim.py`).

Here is traceback 1:
```
This is zim 0.70-rc3
Platform: posix
Locale: en_US UTF-8
FS encoding: utf-8
Python: (3, 6, 7, 'final', 0)
PyGObject: (3, 26, 1)

======= Traceback =======
  File "/home/bit/docs/code/zim-desktop-wiki/zim/actions.py", line 115, in do_activate
    self.__get__(instance, instance.__class__)()
  File "/home/bit/docs/code/zim-desktop-wiki/zim/actions.py", line 98, in func
    return self.func(instance, *args, **kwargs)
  File "/home/bit/docs/code/zim-desktop-wiki/zim/gui/mainwindow.py", line 834, in open_page_back
    self.open_page(record)
  File "/home/bit/docs/code/zim-desktop-wiki/zim/gui/mainwindow.py", line 756, in open_page
    raise AssertionError('Could not save page') # XXX - shouldn't this lead to dialog ?
AssertionError: Could not save page
```

Here is trackback 2:
```
This is zim 0.70-rc3
Platform: posix
Locale: en_US UTF-8
FS encoding: utf-8
Python: (3, 6, 7, 'final', 0)
PyGObject: (3, 26, 1)

======= Traceback =======
  File "/home/bit/docs/code/zim-desktop-wiki/zim/gui/widgets.py", line 2997, in do_response
    destroy = self.do_response_ok()
  File "/home/bit/docs/code/zim-desktop-wiki/zim/gui/uiactions.py", line 571, in do_response_ok
    lines = self.page.dump(format)
  File "/home/bit/docs/code/zim-desktop-wiki/zim/notebook/page.py", line 640, in dump
    tree = self.get_parsetree()
  File "/home/bit/docs/code/zim-desktop-wiki/zim/notebook/page.py", line 556, in get_parsetree
    return self._ui_object.get_parsetree()
  File "/home/bit/docs/code/zim-desktop-wiki/zim/gui/pageview.py", line 5492, in get_parsetree
    self._parsetree = buffer.get_parsetree()
  File "/home/bit/docs/code/zim-desktop-wiki/zim/gui/pageview.py", line 2484, in get_parsetree
    tree = parser.parse(text, partial=tree.ispartial)
  File "/home/bit/docs/code/zim-desktop-wiki/zim/formats/wiki.py", line 486, in parse
    wikiparser(builder, input)
  File "/home/bit/docs/code/zim-desktop-wiki/zim/formats/wiki.py", line 90, in __call__
    self.block_parser(builder, text)
  File "/home/bit/docs/code/zim-desktop-wiki/zim/parser.py", line 398, in __call__
    assert text, 'BUG: processing empty string'
AssertionError: BUG: processing empty string
```

**Comment by Osndok** [comment]
I wonder if this is related to the feature of automatically deleting empty pages (and the page in this example, for a moment, is empty).

Can you try again without the page ever being empty (add the text to the top first, and then delete the original text0?

**Comment by bitinerant** [comment]
With an existing page, I get the same results. In other words, I can reproduce the tracebacks by clearing out an old page (including the title) and typing `a: b` and then navigating to another page. In fact, if I type `a: b` on a line by itself as the first line on a page (above the title and existing text), I don't get a traceback, but the text is silently deleted when I navigate away and then back to the page.

**Comment by Osndok** [comment]
Ahh... yes... I see what's going on...

The line is **being captured into the page headers**. It is either an advanced feature I was not aware of (being able to easily add new headers), or a simple off-by-one error.

I can confirm & reproduce this behavior.

**Comment by bitinerant** [comment]
I don't have titles on all of my pages, and I keep getting bitten by this bug. I guess I have a habit of labeling things with colons, e.g. `See also: ~/Documents/work budgets`, often as the first line. Today, it deleted 3 lines, but fortunately I noticed because I had just opened it, and fortunately I had a backup handy. The only reliable work-around is to see that every page has a title, but I find that cumbersome for pages with just a few lines of text.
